include/boost/math/tools/polynomial.hpp

```diff
@@ -558,59 +562,52 @@ class polynomial :
 
 
 template <class T>
-inline polynomial<T> operator + (const polynomial<T>& a, const polynomial<T>& b)
+inline polynomial<T> operator + (polynomial<T> a, const polynomial<T>& b)
 {
-   polynomial<T> result(a);
-   result += b;
-   return result;
+   a += b;
+   return a;
 }
 
 template <class T>
-inline polynomial<T> operator - (const polynomial<T>& a, const polynomial<T>& b)
+inline polynomial<T> operator - (polynomial<T> a, const polynomial<T>& b)
 {
-   polynomial<T> result(a);
-   result -= b;
-   return result;
+   a -= b;
+   return a;
 }
 
 template <class T>
-inline polynomial<T> operator * (const polynomial<T>& a, const polynomial<T>& b)
+inline polynomial<T> operator * (polynomial<T> a, const polynomial<T>& b)
```

Comment:
I just created the overloads

```
operator+(const polynomial<T>&, const polynomial<T>&)
operator+(const polynomial<T>&, polynomial<T>&&)
operator+(polynomial<T>&&, const polynomial<T>&)
operator+(polynomial<T>&&, polynomial<T>&&)
```
This causes the following code to use the overload with both const lvalue refs
```
polynomial<double> p{1, 2, 3};
polynomial<double> q{1, 2, 3};
auto sum = p + q + p;
```

Which in my opinion is undesirable. Please advise. @jzmaddock